### DISCUSSION TITLE
Mean of all years

### DISCUSSION THREAD

**Discussion by Mohima17** [ProjectPythia/projectpythia.github.io#233] [General]
If I want to get mean of 30years what will be the code with xarray? Dimension would look like (lat, lon, time=1). If I use 'time.year' I get yearly mean and the time dimension is time=30, but I need mean of all year (time=1). Please suggest me how can I get the mean with xarray.

**Comment by brian-rose** [comment]
Hi @Mohima17, we probably need more context (such as some actual example code) to answer your questions properly, but the typical way to take a time mean on an Xarray data array would just look like this:
```
myarray.mean(dim="time")
```

**Reply by Mohima17** [reply]
![image](https://user-images.githubusercontent.com/65295388/165236070-6e2501c1-ffe4-4b9e-b9a8-873e637d047a.png)
How can I get dimension year=1, mean of whole 29 years

**Reply by Mohima17** [reply]
Thank you for your reply.

**Reply by Mohima17** [reply]
My code is:    df.sla.groupby('time.year').mean() / df.groupby('time.year').mean()

**Comment by brian-rose** [comment]
In that screenshot, the time coordinate is called `year`, not `time`. So to average over that dimension you would do
```
[variable name].mean(dim="year")
```

Many xarray operators, including `mean`, accept named dimensions as arguments. But the name must correspond to a dimension that exists in the data array.
